File: game/forest/hurt_branch_animation.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.hurt_branch_talking import HurtBranchTalking
from game_data import GameData
from render_type import RenderType
from state import State


class HurtBranchAnimation(State):
    def __init__(self, context: GameData):
        super().__init__(context)
        self.needs_bottom = RenderType.POST

    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.hugohitlog):
            return HurtBranchTalking
        else:
            return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.sfx_birds, self.context.audio_port)

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.hugohitlog, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtBranchAnimation.ts
/** Port of game/forest/hurt_branch_animation.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { HurtBranchTalking } from './hurtBranchTalking';
import { RenderType } from '../renderType';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class HurtBranchAnimation extends State {
  readonly needsBottom = RenderType.POST;

  constructor(context: GameData) { super(context); }

  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndexFast() >= ForestResources.hugohitlog.length ? HurtBranchTalking : null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.sfxBirds);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.hugohitlog, this.getFrameIndexFast()), 0, 0);
  }
}
